internal/bguploader.go

```diff
@@ -135,23 +158,47 @@ func (b *BgUploader) scanAndProcessFiles() {
 
 }
 
+// speculativelyGenerateNextWalFilenames continually generates and sends the
+// subsequent WAL segment names after b.initialWalSegment to fileChan until ctx
+// is canceled or 100 * b.maxNumUploaded filenames have been generated.
+func (b *BgUploader) speculativelyGenerateNextWalFilenames(ctx context.Context, fileChan chan<- string) {
+	walSegment := b.initialWalSegment
+	for i := 0; i < 100*int(b.maxNumUploaded); i++ {
+		nextWalSegment, err := GetNextWalFilename(walSegment)
+		if err != nil {
+			tracelog.InfoLogger.Printf("failed to parse %s as wal segment name: %v", walSegment, err)
```

Comment:
TODO: this is probably an irrelevant log message for most/all wal-g users. I'll remove this.